### PR TITLE
feat(tests) utilize kong-build-tools functional tests to test this repository

### PR DESCRIPTION
**Pull request by hutchic**

### CODE REVIEW

File: .tavis.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+language: generic
+
+env:
+  matrix:
+    - KONG_TEST_DATABASE=postgres
+    - KONG_TEST_DATABASE=cassandra
+
+install:
+  - make setup_tests
+  - cd kong-build-tools && make setup_tests
```
Comment (hbagdi) [inline .tavis.yml]: This is pretty confusing.
A Makefile in the root repo and then in another repo which we are cloning too.
Is there a way we could simplify this please?

File: run_tests.sh [hunk 2]
```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+
+while [[ "$(kubectl get deployment -n kong | grep 0/1 | wc -l)" != 0 ]]; do
+  echo "waiting for Kong to be ready"
+  sleep 10;
+done
+
+KONG_VERSION=$(kubectl exec -n kong -it $(kubectl get pods -n kong | grep kong | head -n 1 | awk '{print $1}') -- kong version | tr -d '[:space:]')
```
Comment (hbagdi) [inline run_tests.sh]: Why can't you get this from the deployment YAML itself?

Comment (hutchic) [reply run_tests.sh]: the deplayment YAML typically is pinned to `latest` isn't it?

File: .tavis.yml (hunk 1 above)
Comment (hutchic) [reply .tavis.yml]: yeah it's a sad pattern we've ended up with replicating in a few spots.

We could git-submodule which most folks seem opposed to or we could extract the common elements to a bash script and then
```
curl -fsSL https://raw.githubusercontent.com/Kong/kong-build-tools/${KONG_BUILD_TOOLS}/.ci/setup_ci.sh | bash
```

File: run_tests.sh (hunk 2 above)
Comment (hbagdi) [reply run_tests.sh]: Then we should change it to be pinned to a specific version.
Every Kong release, we should verify if there are any updates necessary and then bump it up.

If we want to iterate over this and not make part of this PR, that's cool but we should really be avoiding this hack. Up to you to tackle it in this PR or a later one.

File: Makefile [hunk 3]
```diff
@@ -1,3 +1,15 @@
+KONG_BUILD_TOOLS?=2.0.0
+KONG_TEST_DATABASE?=postgres
+
+setup-kong-build-tools:
+	-rm -rf kong-build-tools; \
+	git clone https://github.com/Kong/kong-build-tools.git; fi
+	cd kong-build-tools; \
+	git reset --hard $(KONG_BUILD_TOOLS); \
```
Comment (hbagdi) [inline Makefile]: nit: should it not be `${KONG_BUILD_TOOLS}`?

File: .tavis.yml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+env:
+  matrix:
+    - K8S_VERSION=v1.15.0 KONG_TEST_DATABASE=postgres
+    - K8S_VERSION=v1.15.0 KONG_TEST_DATABASE=cassandra
+    - K8S_VERSION=v1.14.3 KONG_TEST_DATABASE=postgres
+    - K8S_VERSION=v1.13.6 KONG_TEST_DATABASE=postgres
+matrix:
+  allow_failures:
+    - env: K8S_VERSION=v1.13.6
```
Comment (hbagdi) [inline .tavis.yml]: Why is this allowed?

Comment (hutchic) [reply .tavis.yml]: I found when implementing 1.13.6 tests via k8s manifests that they're not passing reliably. It's on my plate to fix it but having this test would have caught it earlier

**Pull request merged**